### DISCUSSION TITLE
about computing center-pos's detail

### DISCUSSION THREAD

**Discussion by zhjw0927** [STVIR/pysot#414] [General] [answered]
![image](https://user-images.githubusercontent.com/42128459/91399064-e3422c00-e86f-11ea-8626-78697ba4eb22.png)
I found both have '-1' operation in detection and tracking. Why is this?
Thank you!

**Comment by ZhiyuanChen** [comment]
> Why is this?

Because we are tring to get the center position.

**Comment by zhjw0927** [comment]
For example, bbox=[0,0,2,2], the center pos should be (1,1)
If  follow the figure above, pos is (0.5, 0.5)
Where is the gap between them?

**Comment by ZhiyuanChen** [comment]
> For example, bbox=[0,0,2,2], the center pos should be (1,1)
> If follow the figure above, pos is (0.5, 0.5)
> Where is the gap between them?

Because in Computer Science, we start from zero and end at n-1. 
So if you slice a list from x_1 to x_2, the length of which should be x_2 - x_1 -1. 
In your case which the bbox is [0,0,2,2], it's size should be [1, 1, 1, 1], and the center pos corresponding to this matrix should be (0.5, 0.5)

**Comment by ZhiyuanChen** [comment]
> ![image](https://user-images.githubusercontent.com/42128459/91399064-e3422c00-e86f-11ea-8626-78697ba4eb22.png)

<del>Though there is actually a serious bug. This bbox only works when `bbox[0]` and `bbox[2]` is zero since only the second half is being divided by 2. Which means, if we have a box like [10,10,12,12], the center pos will be (15.5, 15.5) which is clearly wrong. </del>

Wait, this box is in the form of ltwh, which means it's actually correct.

**Comment by zhjw0927** [comment]
> > For example, bbox=[0,0,2,2], the center pos should be (1,1)
> > If follow the figure above, pos is (0.5, 0.5)
> > Where is the gap between them?
> 
> Because in Computer Science, we start from zero and end from n-1.
> So if you slice a list from x_1 to x_2, the length of which should be x_2 - x_1 -1.
> In your case which the bbox is [0,0,2,2], it's size should be [1, 1, 1, 1], and the center pos corresponding to this matrix should be (0.5, 0.5)

what does mean that size is [1,1,1,1]? 
bbox=[0,0,2,2], left-top is (0,0), wdith is 2, height is 2 .
Its legth is 2, isn't is fixed?

**Comment by ZhiyuanChen** [comment, marked as answer]
> > > For example, bbox=[0,0,2,2], the center pos should be (1,1)
> > > If follow the figure above, pos is (0.5, 0.5)
> > > Where is the gap between them?
> > 
> > 
> > Because in Computer Science, we start from zero and end from n-1.
> > So if you slice a list from x_1 to x_2, the length of which should be x_2 - x_1 -1.
> > In your case which the bbox is [0,0,2,2], it's size should be [1, 1, 1, 1], and the center pos corresponding to this matrix should be (0.5, 0.5)
> 
> what does mean that size is [1,1,1,1]?
> bbox=[0,0,2,2], left-top is (0,0), wdith is 2, height is 2 .
> Its legth is 2, isn't is fixed?

Sorry you are coorect, the size should be [2,2], and (1,1) refers to the last element of such matrix. The center pos should be (0.5, 0.5)

**Comment by zhjw0927** [comment]
I got it.
slice a list is exclusive.
I came across img or featmap and always wanted to draw  grid, that mislead me.
Thank you!

**Comment by ZhiyuanChen** [comment]
Glad I can help~
